templates/telemetry/telemetry_core_parquet_v3.sql

```diff
@@ -1,12 +1,87 @@
--- Query generated by: templates/unnest_parquet_view.sql.py telemetry_core_parquet_v3 telemetry_derived.telemetry_core_parquet_v3
 CREATE OR REPLACE VIEW
   `moz-fx-data-derived-datasets.telemetry.telemetry_core_parquet_v3` AS
+WITH
+  unioned AS (
+  SELECT
+    *
+  FROM
+    `moz-fx-data-shared-prod.telemetry_stable.core_v2`
+  UNION ALL
+  SELECT
+    *
+  FROM
+    `moz-fx-data-shared-prod.telemetry_stable.core_v3`
+  UNION ALL
+  SELECT
+    *
+  FROM
+    `moz-fx-data-shared-prod.telemetry_stable.core_v4`
+  UNION ALL
+  SELECT
+    *
+  FROM
+    `moz-fx-data-shared-prod.telemetry_stable.core_v5`
+  UNION ALL
+  SELECT
+    *
+  FROM
+    `moz-fx-data-shared-prod.telemetry_stable.core_v6`
+  UNION ALL
+  SELECT
+    *
+  FROM
+    `moz-fx-data-shared-prod.telemetry_stable.core_v7`
+  UNION ALL
+  SELECT
+    *
+  FROM
+    `moz-fx-data-shared-prod.telemetry_stable.core_v8`
+  UNION ALL
+  SELECT
+    *
+  FROM
+    `moz-fx-data-shared-prod.telemetry_stable.core_v9` )
+  --
 SELECT
-  submission_date AS submission_date_s3,
-  * REPLACE (
-    ARRAY(SELECT * FROM UNNEST(accessibility_services.list)) AS accessibility_services,
-    ARRAY(SELECT * FROM UNNEST(experiments.list)) AS experiments,
-    searches.key_value AS searches
-  )
+  DATE(submission_timestamp) AS submission_date_s3,
+  DATE(submission_timestamp) AS submission_date,
+  metadata.uri.app_name,
+  os,
+  STRUCT(
+    document_id,
+    0 AS timestamp,
+    metadata.header.date AS date,
+    metadata.geo.country AS geo_country,
+    metadata.geo.city AS geo_city,
+    metadata.uri.app_build_id AS app_build_id,
+    normalized_channel
+  ) AS metadata,
+  STRUCT(
+    metadata.uri.app_name AS name
+  ) AS application,
+  v,
+  client_id,
+  seq,
+  locale,
+  osversion,
+  device,
+  arch,
+  profile_date,
+  default_search,
+  distribution_id,
+  created,
+  tz,
+  sessions,
+  durations,
+  searches,
+  experiments,
+  flash_usage,
+  campaign,
+  campaign AS campaign_id,
+  default_browser,
+  show_tracker_stats_share,
+  accessibility_services,
+  metadata.uri.app_version AS metadata_app_version,
+  bug_1501329_affected
 FROM
-  `moz-fx-data-derived-datasets.telemetry_derived.telemetry_core_parquet_v3`
+  unioned
```

Comment:
Didn't you mention in a previous PR that `UNION ALL` breaks partition filtering? I'm pretty sure we can use wildcard tables instead:
```suggestion
  `moz-fx-data-shared-prod.telemetry_stable.core_v*`
WHERE
  _TABLE_SUFFIX != '1'
```

I tested this, and it works.

I don't understand why, but `SELECT * FROM moz-fx-data-shar-nonprod-efed.telemetry_live.core_v*` is limited to columns in **all** of the wildcard tables, while `SELECT * FROM moz-fx-data-shar-nonprod-efed.telemetry_stable.core_v*` gets columns in **any** of the wildcard tables.

copying the tables to new datasets in `gcp-ingestion-dev` had the same results as `telemetry_stable`, so perhaps there was a recent change to bigquery implementation that only applies to new datasets or tables and makes this work the way we need.

Reply:
> Didn't you mention in a previous PR that UNION ALL breaks partition filtering?

I don't recall that, but wildcard looks good. I had forgotten that we could exclude v1 via table suffix.

Reply:
Perhaps the fact that the `_stable` tables are empty while the live tables have content is relevant to how the wildcard table matching works.

Reply:
> Currently, cached results are not supported for queries against multiple tables using a wildcard

This makes me hesitant to use table wildcards in views.

Reply:
Another interesting tidbit that could explain differences in behavior:

> BigQuery uses the schema for the most recently created table that matches the wildcard as the schema for the wildcard table. If the schema is inconsistent across tables matched by the wildcard table, BigQuery returns an error

I don't know whether filtering on table prefix affects the chosen schema.

Reply:
> BigQuery uses the schema for the most recently created table that matches the wildcard as the schema for the wildcard table

THAT EXPLAINS IT! if `core_v1` were created after `core_v10`, then wildcard would get the wrong schema, and we'd have issues.

> If the schema is inconsistent across tables matched by the wildcard table, BigQuery returns an error

based on tests, this only matters if the schema of columns actually collected is incompatible, not if it's merely different:

```bash
bq mk gcp-ingestion-dev:test

# v2 schema second
bq query -q 'CREATE OR REPLACE TABLE `gcp-ingestion-dev.test.core_v1` AS SELECT 1 AS a'
bq query -q 'CREATE OR REPLACE TABLE `gcp-ingestion-dev.test.core_v2` AS SELECT 2 AS a, 3 AS b'
bq query -q --format=json -n9 'SELECT * FROM `gcp-ingestion-dev.test.core_v*`'
# [{"a":"2","b":"3"},{"a":"1","b":null}]

# v1 schema second
bq query -q 'CREATE OR REPLACE TABLE `gcp-ingestion-dev.test.core_v1` AS SELECT 1 AS a'
bq query -q --format=json -n9 'SELECT * FROM `gcp-ingestion-dev.test.core_v*`'
# [{"a":"2"},{"a":"1"}]

# v3 schema incompatible
bq query -q 'CREATE OR REPLACE TABLE `gcp-ingestion-dev.test.core_v3` AS SELECT 4 AS a, "" AS b'
bq query -q --format=json -n9 'SELECT * FROM `gcp-ingestion-dev.test.core_v*`'
# Error in query string: Error processing job '...': Cannot read field 'b' of type INT64 as STRING

# v3 schema only compatible columns
bq query -q --format=json -n3 'SELECT a FROM `gcp-ingestion-dev.test.core_v*`'
# [{"a":"4"},{"a":"2"},{"a":"1"}]

# v3 schema only compatible tables
bq query -q --format=json -n3 'SELECT * FROM `gcp-ingestion-dev.test.core_v*` WHERE _TABLE_SUFFIX = "3"'
# [{"a":"4","b":""}]
```

Reply:
> I don't know whether filtering on table prefix affects the chosen schema.

It does not

Reply:
in conclusion: I think we should use wildcard tables, and require that our tooling creates new tables in order, so that wildcard tables always get the latest version schema

Reply:
> in conclusion: I think we should use wildcard tables, and require that our tooling creates new tables in order, so that wildcard tables always get the latest version schema

Core ping feels like a special case that perhaps doesn't warrant special support in tooling. I'd expect that when a new version of a schema is published, it's backwards-incompatible in some way such that a wildcard-based union would likely break. I don't really understand the history of core ping, but it looks like we mistakenly created new versions simply to add new fields, so we were able to backport those fields to previous versions and have them all be compatible.

If I'm correct in characterizing core ping as a special case, I'd prefer to leave the explicit union and not throw more effort down making wildcards work.